src/legacy/core_plugins/metrics/public/components/index_pattern.js

```diff
@@ -18,124 +18,181 @@
         </EuiFlexItem>
         <EuiFlexItem>
           <EuiFormRow
             id={htmlId('timeField')}
-            label={
-              <FormattedMessage id="tsvb.indexPattern.timeFieldLabel" defaultMessage="Time field" />
-            }
-            fullWidth
+            label={i18n.translate('tsvb.indexPattern.timeFieldLabel', {
+              defaultMessage: 'Time field',
+            })}
           >
             <FieldSelect
               data-test-subj="metricsIndexPatternFieldsSelect"
               restrict={RESTRICT_FIELDS}
               value={model[timeFieldName]}
-              disabled={props.disabled}
+              disabled={disabled}
               onChange={handleSelectChange(timeFieldName)}
               indexPattern={model[indexPatternName]}
               fields={fields}
               placeholder={isDefaultIndexPatternUsed ? model.default_timefield : undefined}
-              fullWidth
             />
           </EuiFormRow>
         </EuiFlexItem>
         <EuiFlexItem grow={false}>
           <EuiFormRow
             id={htmlId('interval')}
-            label={
-              <FormattedMessage id="tsvb.indexPattern.intervalLabel" defaultMessage="Interval" />
-            }
-            helpText={
-              <FormattedMessage
-                id="tsvb.indexPattern.intervalHelpText"
-                defaultMessage="Examples: auto, 1m, 1d, 7d, 1y, >=1m"
-                description="auto, 1m, 1d, 7d, 1y, >=1m are required values and must not be translated."
-              />
-            }
+            label={i18n.translate('tsvb.indexPattern.intervalLabel', {
+              defaultMessage: 'Interval',
+            })}
+            helpText={i18n.translate('tsvb.indexPattern.intervalHelpText', {
+              defaultMessage: 'Examples: auto, 1m, 1d, 7d, 1y, >=1m',
+              description:
+                'auto, 1m, 1d, 7d, 1y, >=1m are required values and must not be translated.',
+            })}
           >
             <EuiFieldText
-              disabled={props.disabled}
+              disabled={disabled || isEntireTimeRangeActive(model, isTimeSeries)}
               onChange={handleTextChange(intervalName, 'auto')}
               value={model[intervalName]}
               placeholder={'auto'}
             />
           </EuiFormRow>
         </EuiFlexItem>
         <EuiFlexItem grow={false}>
-          <EuiFormLabel>
-            <FormattedMessage
-              id="tsvb.indexPattern.dropLastBucketLabel"
-              defaultMessage="Drop last bucket?"
+          <EuiFormRow
+            id={htmlId('dropLastBucket')}
+            label={i18n.translate('tsvb.indexPattern.dropLastBucketLabel', {
+              defaultMessage: 'Drop last bucket?',
+            })}
+          >
+            <YesNo
+              value={model[dropBucketName]}
+              name={dropBucketName}
+              onChange={onChange}
+              disabled={isEntireTimeRangeActive(model, isTimeSeries) || disabled}
```

Comment:
Just for being consistent with the code above, could we also change the order of `disabled` and `isEntireTimeRangeActive()`?

Reply:
Changed